runware/base.py

```diff
@@ -1367,6 +1368,174 @@ def check(resolve: Callable, reject: Callable, *args: Any) -> bool:
 
             raise RunwareAPIError({"message": str(e)})
 
+    async def videoInference(self, requestVideo: IVideoInference) -> List[IVideo]:
+        try:
+            await self.ensureConnection()
+            return await asyncRetry(lambda: self._requestVideo(requestVideo))
+        except Exception as e:
+            raise e
+
+    async def _requestVideo(self, requestVideo: IVideoInference) -> List[IVideo]:
+        await self._processVideoImages(requestVideo)
+        requestVideo.taskUUID = requestVideo.taskUUID or getUUID()
+        request_object = self._buildVideoRequest(requestVideo)
+        await self.send([request_object])
+        return await self._handleInitialVideoResponse(requestVideo.taskUUID, requestVideo.numberResults)
+
+    async def _processVideoImages(self, requestVideo: IVideoInference) -> None:
+        if requestVideo.frameImages:
+            processed_frame_images = []
+            for frame_item in requestVideo.frameImages:
+                if isinstance(frame_item, IFrameImage):
+                    frame_item.inputImages = await process_image(frame_item.inputImage)
+                    processed_frame_images.append(frame_item)
+            requestVideo.frameImages = processed_frame_images
```

Comment:
what's even the idea of async approach if you are processing it synchronously here

Reply:
use asyncio.gather instead in order to access real async